client/mapHandler.cpp

```diff
@@ -536,6 +536,8 @@ void CMapHandler::CMapWorldViewBlitter::drawTileOverlay(SDL_Surface * targetSurf
 	for(auto & object : objects)
 	{
 		const CGObjectInstance * obj = object.obj;
+		if(!obj)

```

Comment:
Interesting case. What does it mean?

Reply:
Multithreading problem - attempt to dereference object which was deleted. General rule - all ptrs has to be checked for nullptr. If it's not expected, we should use references

Reply:
Usually multithreading problems are solved by proper locking. Any additional details on bug mechanics. Should not object be erased first from objects list and deleted afterwards?

Reply:
Why it should be erased from the list? List is valid, but object has an obj field which is a pointer and could be nullptr, there is no restrictions

Reply:
I agree that maybe root cause is in another place, but rendering must not crash the game